lib/Etsy/StatsD.pm

```diff
@@ -114,19 +185,16 @@ sub send {
 
 	#failures in any of this can be silently ignored
 	my $count  = 0;
-	my $socket = $self->{socket};
-	while ( my ( $stat, $value ) = each %$sampled_data ) {
-		_send_to_sock($socket, "$stat:$value\n");
-		++$count;
-	}
+	foreach my $socket ( @{ $self->{sockets} } ) {
+        # Using foreach/keys because each iterator is unreliable when nested
```

Comment:
Nested iterators should be fine as long as they're iterating different collections, since the iterator state is stored in the collection.  This could have some performance ramifications, since if a large set of data is passed in, this should perform an implicit copy of the key set which would've been avoided with the iterator.